### PR TITLE
Synchronously update nginx for the first update

### PR DESCRIPTION
**Pull request by chbatey**
Otherwise ingress can be unavailble for up to 30s

### CODE REVIEW

File: nginx/nginx.go [hunk 1]
```diff
@@ -275,7 +275,11 @@ func (lb *nginxLoadBalancer) Update(entries controller.IngressUpdate) error {
 		return fmt.Errorf("unable to update nginx: %v", err)
 	}
 
-	if updated {
+	if updated && !lb.initialUpdateApplied.Get() {
+		log.Info("Initial update. Signalling nginx synchronously.")
+		lb.signaller.sighup(lb.cmd.Process)
+		lb.initialUpdateApplied.Set(true)
+	} else if updated {
```
Comment (jsravn) [inline nginx/nginx.go]: Any reason to not nest the conditional? It's a little confusing as-is I think.

```go
if updated {
  if !lb.initialUpdateApplied.Get() {
     updateSync()
  } else {
     updateAsync()
  }
}
```

Comment (chbatey) [inline nginx/nginx.go]: can do

**Comment by chbatey** [comment]
Hrmm I agree. Q reasonable change given "updaters" are currently independent

**Comment by chbatey** [comment]
Pushed a second commit @jsravn that puts the register ELBs in the first update. I didn't use `sync.Once` as that doesn't appear to allow me to return a value which makes sense.

File: elb/elb.go [hunk 2]
```diff
@@ -51,6 +54,12 @@ type elb struct {
 	instanceID          string
 	elbs                map[string]LoadBalancerDetails
 	registeredFrontends int
+	initialised         initialised
+}
+
+type initialised struct {
+	sync.Mutex
+	done bool
 }
```
Comment (jsravn) [inline elb/elb.go]: You can use the atomic bool inside the util package.

**Pull request merged**